# Question 1  
  
(10 Marks) Carry out a principal component analysis on the temperature data of Table 7.2. There are 11 variables and 46 obsevations. Use both $\textbf{S}$ and $\textbf{R}$.

In [8]:
data <- read.table(file = "/Users/alexwiigs/Documents/Code/STAT 553 Assigmnents/Assignment 5/T7_2_TEMPERAT.DAT")
data <- as.matrix(data)
colnames(data) <- c("Max Air Temp", "Min Air Temp", "Avg Air Temp", "Max Soil Temp", "Min Soil Temp", "Avg Soil Temp", "Max Humidity", "Min Humidity", "Avg Humidity", "Total Wind", "Evaporation")
head(data)
n = 46

Max Air Temp Min Air Temp Avg Air Temp Max Soil Temp Min Soil Temp
[1,] 84           65           147          85            59           
[2,] 84           65           149          86            61           
[3,] 79           66           142          83            64           
[4,] 81           67           147          83            65           
[5,] 84           68           167          88            69           
[6,] 74           66           131          77            67           
     Avg Soil Temp Max Humidity Min Humidity Avg Humidity Total Wind
[1,] 151           95           40           398          273       
[2,] 159           94           28           345          140       
[3,] 152           94           41           368          318       
[4,] 158           94           50           406          282       
[5,] 180           93           46           379          311       
[6,] 147           96           73           478          446       
     Evaporation
[1,] 30         
[2,] 34         
[3,] 33         
[4,] 26         
[5,] 41         
[6,]  4

a) Show the eigenvalues and percent of variance explained.

**Using $S$:**

In [40]:
S <- unname(1/(n-1) * ( t(data) %*% data - t(data) %*% matrix(1, nrow= 46, ncol = 46) %*% data / n ) )
round(S,2)

[,1]    [,2]   [,3]    [,4]   [,5]   [,6]    [,7]   [,8]    [,9]   
 [1,]   55.68  16.49  117.55  26.98  10.38   97.73  -1.56  -42.36 -128.27
 [2,]   16.49  10.86   61.67  14.03   8.23   56.48  -0.66  -10.80  -44.40
 [3,]  117.55  61.67  402.70  92.95  43.46  365.27  -3.86 -106.89 -387.72
 [4,]   26.98  14.03   92.95  25.66  10.60   92.81  -0.63  -26.67  -94.33
 [5,]   10.38   8.23   43.46  10.60  13.44   59.93  -0.54    8.46   -4.36
 [6,]   97.73  56.48  365.27  92.81  59.93  438.25  -1.02  -59.94 -313.47
 [7,]   -1.56  -0.66   -3.86  -0.63  -0.54   -1.02   1.45    1.90    9.97
 [8,]  -42.36 -10.80 -106.89 -26.67   8.46  -59.94   1.90  106.20  271.65
 [9,] -128.27 -44.40 -387.72 -94.33  -4.36 -313.47   9.97  271.65  885.63
[10,] -209.10  14.93 -294.55 -67.93 224.58  386.82 -26.38  597.67  970.97
[11,]   61.37  25.99  201.87  53.61  17.58  217.06  -3.24  -96.85 -355.39
      [,10]    [,11]  
 [1,]  -209.10   61.37
 [2,]    14.93   25.99
 [3,]  -294.55  201.87
 [4,]   -67.93   53.61
 [5,]   224.58   17.58
 [6,]   386.82  217.06
 [7,]   -26.38   -3.24
 [8,]   597.67  -96.85
 [9,]   970.97 -355.39
[10,] 22227.16   94.59
[11,]    94.59  214.06

Using $S$ we can calculate the percent of variance explained as...

In [39]:
percentage = matrix(nrow = 1, ncol = 11)
for (i in 1:11) { percentage[i] = eigen(S)$values[i] / sum(eigen(S)$values)  }
displayS <- rbind(eigen(S)$values, percentage)
row.names(displayS) <- c("Eigenvalues:", "Variance Explained:")
round(displayS,4)


[,1]       [,2]      [,3]     [,4]    [,5]    [,6]   
Eigenvalues:        22303.4976 1590.6789 358.0457 63.3665 29.3270 17.1149
Variance Explained:     0.9148    0.0652   0.0147  0.0026  0.0012  0.0007
                    [,7]    [,8]   [,9]   [,10]  [,11] 
Eigenvalues:        12.7478 2.8330 1.9069 0.8769 0.7028
Variance Explained:  0.0005 0.0001 0.0001 0.0000 0.0000

The first component accounts for about 91 percent and the variance. 

**Using $R$:**

Calculating $R$ and repeating the same procedure above we get:

In [36]:
D <- diag(sqrt(diag(S)))
R <- solve(D) %*% S %*% solve(D)
round(R,2)

[,1]  [,2]  [,3]  [,4]  [,5]  [,6]  [,7]  [,8]  [,9]  [,10] [,11]
 [1,]  1.00  0.67  0.79  0.71  0.38  0.63 -0.17 -0.55 -0.58 -0.19  0.56
 [2,]  0.67  1.00  0.93  0.84  0.68  0.82 -0.17 -0.32 -0.45  0.03  0.54
 [3,]  0.79  0.93  1.00  0.91  0.59  0.87 -0.16 -0.52 -0.65 -0.10  0.69
 [4,]  0.71  0.84  0.91  1.00  0.57  0.88 -0.10 -0.51 -0.63 -0.09  0.72
 [5,]  0.38  0.68  0.59  0.57  1.00  0.78 -0.12  0.22 -0.04  0.41  0.33
 [6,]  0.63  0.82  0.87  0.88  0.78  1.00 -0.04 -0.28 -0.50  0.12  0.71
 [7,] -0.17 -0.17 -0.16 -0.10 -0.12 -0.04  1.00  0.15  0.28 -0.15 -0.18
 [8,] -0.55 -0.32 -0.52 -0.51  0.22 -0.28  0.15  1.00  0.89  0.39 -0.64
 [9,] -0.58 -0.45 -0.65 -0.63 -0.04 -0.50  0.28  0.89  1.00  0.22 -0.82
[10,] -0.19  0.03 -0.10 -0.09  0.41  0.12 -0.15  0.39  0.22  1.00  0.04
[11,]  0.56  0.54  0.69  0.72  0.33  0.71 -0.18 -0.64 -0.82  0.04  1.00

In [38]:
percentage = matrix(nrow = 1, ncol = 11)
for (i in 1:11) { percentage[i] = eigen(R)$values[i] / sum(eigen(R)$values)  }
displayR <- rbind(eigen(R)$values, percentage)
row.names(displayR) <- c("Eigenvalues:", "Variance Explained:")
round(displayR,4)

[,1]   [,2]   [,3]   [,4]   [,5]   [,6]   [,7]   [,8]  
Eigenvalues:        6.0202 2.1193 1.1303 0.7600 0.3554 0.2593 0.1221 0.1105
Variance Explained: 0.5473 0.1927 0.1028 0.0691 0.0323 0.0236 0.0111 0.0100
                    [,9]   [,10]  [,11] 
Eigenvalues:        0.0598 0.0422 0.0209
Variance Explained: 0.0054 0.0038 0.0019

The first component accounts for about 54% of the data and the second accounts for about 19%. This is a very different result than using $S$.

b) Based on the average eigenvalue, decide how many components to retain.

We calculate the average eigenvalue and include any components whose accompanying eigenvalues are above the average.

**Using $S$:**

In [43]:
lambda_avg <- sum(eigen(S)$values) / 11
round(lambda_avg,2)

[1] 2216.46

The average eigenvalue is $2216.46$. By the average eigenvalue guideline we should only include one component.

**Using $R$:**

In [46]:
lambda_avg <- sum(eigen(R)$values) / 11
round(lambda_avg,2)

[1] 1

The average eigenvalue is $1$. By this guileline we should retain 3 components.

c) Can you interpret the components of $\textbf{S}$?

The components of $S$ can be interpreted as the natural axes of the swarm of points in the mass of data calculated. We have rotated the mass so that the center is at the origin and the new variables (components) will be uncorrelated. In the case of $S$, we see that the first two eiganvalues are significantly larger than the the rest, meaning they account for the vast majority of the variance in the observations. 

Principal components are not scale invariate. In the case of $S$, the swarm of points have been elongated around the first principal component, which likely represents the variable with a much larger variance than the rest. This is why our criteoren only included one component for $S$. In this case it might be better to use $R$.